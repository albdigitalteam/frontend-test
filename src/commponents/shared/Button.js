import React from 'react';
import { ActivityIndicator } from 'react-native'
import styled, { css } from 'styled-components'
import { Text } from './'
import { colors } from '../../utils'

const Button = styled.TouchableOpacity`
  padding: 16px 16px;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  ${props => props.icon && css`padding: 16px;`}
  ${props => props.width && css`width: ${props.width}%;`}
  ${props => props.bgColor && css`background-color: ${props.bgColor};`}
  ${props => props.full && css`
    width: 100%;
    align-items: center;
  `}
  ${props => props.outline && css`
    border-width: 1px;
    border-color: ${props.borderColor};
    background-color: transparent;
  `}
  ${props => props.disabled && css`
    opacity: 0.4;
  `}
  ${props => props.rounded && css`
    border-radius: 4px;
  `}
  ${props => props.float && css`
    border-radius: 30px;
    position: absolute;
    top: 15px;
    right: 15px;
  `}
  ${props => props.pill && css`
    border-radius: 32px;
    padding: 4px 12px;
  `}
  ${props => props.shadow && css`
    shadow-color: gray;
    shadow-offset: 0px 0px;
    shadow-opacity: 0.5;
    shadow-radius: 4px;
  `}
`

export default ({ ...props }) => (
  <Button {...props}>
    {props.loading && <ActivityIndicator color={props.spinnerColor} />}
    {!props.loading && 
      <Text 
        weight={props?.labelWeight ? props.labelWeight : 'normal'}
        color={props?.labelColor ? props.labelColor : colors.black}
        size={props?.labelSize ? props.labelSize : 16} >
          {props.label}
      </Text>
    }
  </Button>
)
