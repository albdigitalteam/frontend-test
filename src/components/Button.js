import React, { memo } from 'react';
import { ActivityIndicator } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import Label from '@/components/Label';

import colors from '@/theme/colors';

import { handlePixels } from '@/helpers/functions/utils';

const renderButton = (
  testID,
  width,
  marginTop,
  marginBottom,
  marginLeft,
  marginRight,
  paddingTop,
  paddingBottom,
  paddingLeft,
  paddingRight,
  backgroundColor,
  color,
  borderWidth,
  modal,
  gradient,
  icon,
  leftIcon,
  onPress,
  isLoading,
  loadingColor,
  disabled,
  children
) => (
  <StyledTouchableOpacity
    testID={testID}
    width={width}
    marginTop={marginTop}
    marginBottom={marginBottom}
    marginLeft={marginLeft}
    marginRight={marginRight}
    paddingTop={paddingTop}
    paddingBottom={paddingBottom}
    paddingLeft={paddingLeft}
    paddingRight={paddingRight}
    backgroundColor={backgroundColor}
    color={color}
    borderWidth={borderWidth}
    modal={modal}
    gradient={gradient}
    icon={icon}
    leftIcon={leftIcon}
    onPress={onPress}
    disabled={disabled}
  >
    {isLoading ? (
      <ActivityIndicator size='small' color={loadingColor} />
    ) : (
      <StyledButtonContainer>
        {icon && <StyledRightIcon source={icon} />}
        <Label
          color={color || '#fff'}
          fontWeight={700}
          textTransform='uppercase'
        >
          {children}
        </Label>
        {leftIcon && <StyledLeftIcon source={leftIcon} />}
      </StyledButtonContainer>
    )}
  </StyledTouchableOpacity>
);

const Button = ({
  testID,
  width,
  marginTop,
  marginBottom,
  marginLeft,
  marginRight,
  paddingTop,
  paddingBottom,
  paddingLeft,
  paddingRight,
  backgroundColor,
  color,
  borderWidth,
  modal,
  isLoading,
  loadingColor,
  gradient,
  icon,
  leftIcon,
  onPress,
  children,
  disabled
}) => (
  <>
    <StyledLinearGradient
      width={width}
      marginTop={marginTop}
      marginBottom={marginBottom}
      marginLeft={marginLeft}
      marginRight={marginRight}
      backgroundColor={backgroundColor}
      color={color}
      onPress={onPress}
      colors={disabled ? colors.GRADIENT_GRAY : colors.GRADIENT_BADGE}
      useAngle
      angle={276}
    >
      {renderButton(
        testID,
        width,
        marginTop,
        marginBottom,
        marginLeft,
        marginRight,
        paddingTop,
        paddingBottom,
        paddingLeft,
        paddingRight,
        backgroundColor,
        color,
        borderWidth,
        modal,
        gradient,
        icon,
        leftIcon,
        onPress,
        isLoading,
        loadingColor,
        disabled,
        children
      )}
    </StyledLinearGradient>
  </>
);

Button.defaultProps = {
  testID: null,
  width: '100%',
  marginTop: 0,
  marginBottom: 0,
  marginLeft: 0,
  marginRight: 0,
  paddingTop: 14,
  paddingBottom: 14,
  paddingLeft: 0,
  paddingRight: 0,
  backgroundColor: '#fff',
  color: '#fff',
  borderWidth: '0px',
  loadingColor: '#fff',
  gradient: false,
  icon: null,
  leftIcon: null,
  disabled: false,
  onPress: () => {},
  isLoading: false,
  modal: false,
  children: null
};

Button.propTypes = {
  testID: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginTop: PropTypes.number,
  marginBottom: PropTypes.number,
  marginLeft: PropTypes.number,
  marginRight: PropTypes.number,
  paddingTop: PropTypes.number,
  paddingBottom: PropTypes.number,
  paddingLeft: PropTypes.number,
  paddingRight: PropTypes.number,
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  borderWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  loadingColor: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  gradient: PropTypes.bool,
  icon: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  leftIcon: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  disabled: PropTypes.bool,
  onPress: PropTypes.func,
  isLoading: PropTypes.bool,
  modal: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.string,
    PropTypes.element
  ])
};

const StyledTouchableOpacity = styled.TouchableOpacity`
  width: 100%;
  min-width: 100px;
  justify-content: center;
  align-items: center;
  margin-top: ${({ marginTop }) => handlePixels(marginTop)};
  margin-bottom: ${({ marginBottom }) => handlePixels(marginBottom)};
  margin-right: ${({ marginRight }) => handlePixels(marginRight)};
  margin-left: ${({ marginLeft }) => handlePixels(marginLeft)};
  padding-top: ${({ paddingTop }) => handlePixels(paddingTop)};
  padding-bottom: ${({ paddingBottom }) => handlePixels(paddingBottom)};
  padding-right: ${({ paddingRight }) => handlePixels(paddingRight)};
  padding-left: ${({ paddingLeft }) => handlePixels(paddingLeft)};
  background-color: transparent;
`;

const StyledRightIcon = styled.Image`
  margin-right: 12px;
`;

const StyledLeftIcon = styled.Image`
  margin-left: 12px;
`;

const StyledButtonContainer = styled.View`
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const StyledLinearGradient = styled(LinearGradient)`
  min-width: 100px;
  min-height: 48px;
  justify-content: center;
  align-items: center;
  margin-top: ${({ marginTop }) => handlePixels(marginTop)};
  margin-bottom: ${({ marginBottom }) => handlePixels(marginBottom)};
  margin-right: ${({ marginRight }) => handlePixels(marginRight)};
  margin-left: ${({ marginLeft }) => handlePixels(marginLeft)};
  border-radius: 8px;
`;

export default memo(Button);
