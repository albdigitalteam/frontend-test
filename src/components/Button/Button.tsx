import React from 'react';
import {
  ActivityIndicator,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import {ColorProps, createBox, useTheme} from '@shopify/restyle';

import Text from '@components/Text/Text';
import {Theme} from '@theme/theme';

type TouchableProps = TouchableOpacityProps & {
  children?: React.ReactNode;
};

const BaseButton = createBox<Theme, TouchableProps>(TouchableOpacity);

type Props = React.ComponentProps<typeof BaseButton> &
  ColorProps<Theme> & {
    label: string;
    isLoading?: boolean;
  };

export default function Button({label, isLoading, color, ...props}: Props) {
  const theme = useTheme<Theme>();

  return (
    <BaseButton
      flexDirection="row"
      {...props}
      disabled={isLoading}
      alignItems="center"
      justifyContent="center">
      <Text
        variant="body"
        color={color}
        textAlign="center"
        marginRight={isLoading ? 's' : undefined}>
        {label}
      </Text>
      {isLoading ? (
        <ActivityIndicator color={theme.colors.purplePrimary} />
      ) : null}
    </BaseButton>
  );
}
