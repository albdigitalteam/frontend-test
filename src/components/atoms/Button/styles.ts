import styled from 'styled-components';

export const Container = styled.button`
  background-color: rgb(55, 95, 155);
  width: 150px;
  color: #fff;
  border-radius: 30px;
  border: none;
  cursor: pointer;
  padding: 15px;
`;

export const Image = styled.img``;
