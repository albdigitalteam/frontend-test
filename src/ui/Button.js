import React from 'react'
import { Text, TouchableOpacity, StyleSheet } from 'react-native'
import * as COLORS from '../helpers/colors'

export default function Button({label, onPress, style}) {
    return (
        <TouchableOpacity onPress={onPress} style={[styles.container, style]}>
            <Text style={styles.label}>{label}</Text>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    container: {
        borderRadius: 6,
        backgroundColor: COLORS.button,
        padding: 12,
        margin: 10,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    },
    label: {
        fontSize: 14,
        color: COLORS.textInvert,
        textTransform: 'uppercase'
    }
})