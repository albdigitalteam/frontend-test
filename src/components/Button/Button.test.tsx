import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from './Button';

describe('test button component', () => {
  it('render and verify text content', () => {
    const testLabel = 'Criar post';
    const { getByText } = render(
      <Button label={testLabel} onPress={jest.fn} />,
    );

    expect(getByText(testLabel)).toBeTruthy();
  });
});

describe('test click button component', () => {
  it('render and verify onPress content', () => {
    const onPress = jest.fn();
    const testLabel = 'Criar post';
    const { getByText } = render(
      <Button label={testLabel} onPress={onPress} />,
    );

    fireEvent.press(getByText(testLabel));
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});

describe('snapshopt test button component', () => {
  it('render button', () => {
    const testLabel = 'Criar post';
    const tree = render(
      <Button label={testLabel} onPress={jest.fn} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
