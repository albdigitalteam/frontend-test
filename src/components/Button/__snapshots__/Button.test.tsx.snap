// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshopt test button component render button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    Criar post
  </Text>
</View>
`;
